server/src/main/java/org/elasticsearch/action/ingest/SimulatePipelineRequest.java

```diff
@@ -126,6 +128,16 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        if (source != null) {
```

Comment:
can the API even work when the source is null? maybe we should rather add this check to validate instead?

Reply:
Looking at the constructor I don't think this will occur. Except if you create it with the blank instance constructor (which only happens in tests). I think we can remove this check if we ignore that constructor.

Reply:
makes sense, the default constructor is there only for serialization, it will be used to create an instance and then call readFrom on it.